tools/behavior_simulator/src/server.rs

```diff
@@ -0,0 +1,173 @@
+    // subscribed outputs immediately after subscribing
+    let mut interval = interval(Duration::from_secs(1));
+
+    loop {
+        select! {
+            _ = parameters_changed.notified() => { }
+            _ = interval.tick() => { }
+            _ = keep_running.cancelled() => {
+                break
+            }
+        }
+
+        let parameters = parameters_reader.next();
+
+        {
+            let mut outputs = outputs_writer.next();
+            outputs.main_outputs.frame_count = frames.len();
+            outputs.main_outputs.databases = frames[parameters.selected_frame].robots.clone();
+        }
+        outputs_changed.notify_waiters();
+
+        {
+            let mut control = control_writer.next();
+            *control = parameters
+                .selected_robot
+                .try_into()
+                .ok()
+                .and_then(|player_number| {
+                    frames[parameters.selected_frame].robots.get(&player_number)
+                })
+                .cloned()
+                .unwrap_or_default();
+        }
+        control_changed.notify_waiters();
+    }
+}
+
+pub fn run(keep_running: CancellationToken, scenario_file: impl AsRef<Path>) -> Result<()> {
+    let communication_server = communication::server::Runtime::<Configuration>::start(
+        Some("[::]:1337"),
+        "tools/behavior_simulator",
+        "behavior_simulator".to_string(),
+        "behavior_simulator".to_string(),
+        2,
+        keep_running.clone(),
+    )?;
+
+    let (outputs_writer, outputs_reader) =
+        multiple_buffer_with_slots([Default::default(), Default::default(), Default::default()]);
+
+    let outputs_changed = Arc::new(Notify::new());
+    let (subscribed_outputs_writer, _subscribed_outputs_reader) =
+        multiple_buffer_with_slots([Default::default(), Default::default(), Default::default()]);
+
+    communication_server.register_cycler_instance(
+        "BehaviorSimulator",
+        outputs_changed.clone(),
+        outputs_reader,
+        subscribed_outputs_writer,
+    );
+
+    let (control_writer, control_reader) =
+        multiple_buffer_with_slots([Default::default(), Default::default(), Default::default()]);
+
+    let control_changed = Arc::new(Notify::new());
+    let (subscribed_control_writer, _subscribed_control_reader) =
+        multiple_buffer_with_slots([Default::default(), Default::default(), Default::default()]);
+    communication_server.register_cycler_instance(
+        "Control",
+        control_changed.clone(),
+        control_reader,
+        subscribed_control_writer,
+    );
+
+    let mut simulator = Simulator::new();
+    simulator.execute_script(scenario_file)?;
+
+    let start = Instant::now();
+    let frames = simulator.run().context("failed to run simulation")?;
+    let duration = Instant::now() - start;
+    println!("Took {:.2} seconds", duration.as_secs_f32());
+
+    let runtime = tokio::runtime::Runtime::new()?;
```

Comment:
IMHO the runtime should be an implementation detail of the server.

Is there a reason to use the builder if I don't have specific requirements?

Reply:
I think there is nothing against using `#[tokio::main]`. This is an asynchronous process, we can fully commit to the async runtime. What do you think?

Reply:
When I try to use `tokio::main`, it crashes with the following error message:
```
Message:  Cannot block the current thread from within a runtime. This happens because a function attempted to block the current thread while the thread is being used to drive asynchronous tasks.
Location: /home/mystery/hulks/hulk-behavior-simulator/crates/communication/src/server/runtime.rs:177
```
Not sure how to fix.